docs/runner.md

```diff
@@ -1,6 +1,158 @@
 ## Runner
 
-The runner module aims to help users to start training with less code, while stays
-flexible and configurable.
+The runner class is designed to manage the training. It eases the training process with less code demanded from users while staying flexible and configurable. The main features are as listed:
 
-Documentation and examples are still on going.
+- Support `EpochBasedRunner` and `IterBasedRunner` for different scenarios.
+- Support customized workflow to allow switching between different modes while training. Currently, supported modes are train and val.
+- Enable extensibility through various Hooks, including hooks defined in MMCV and customized ones.
+
+## EpochBasedRunner
+
+As its name indicates, workflow in `EpochBasedRunner` should be set based on epochs. For example, [('train', 2), ('val', 1)] means running 2 epochs for training and 1 epoch for validation, iteratively. And each epoch may contain multiple iters. Currently, MMDetection uses `EpochBasedRunner` by default.
+
+Let's take a look at its core logic:
+
+```python
+# the condition to stop training
+while curr_epoch < max_epochs:
+    # traverse the workflow.
+    # e.g. workflow = [('train', 2), ('val', 1)]
+    for i, flow in enumerate(workflow):
+        # mode(e.g. train) determines which function to run
+        mode, epochs = flow
+        epoch_runner = getattr(self, mode)
+        # execute the corresponding function
+        for _ in range(epochs):
+            epoch_runner(data_loaders[i], **kwargs)
+```
+
+Currently, we support 2 modes: train and val. Let's take a look at their core logic:
+
+```python
+# Currently, epoch_runner could be either train or val
+def epoch_runner(self, data_loader, **kwargs):
+    # traverse the dataset and get batch data for 1 epoch
+    for i, data_batch in enumerate(data_loader):
+        self.call_hook('before_train(val)_iter')
+        self.run_iter(data_batch, train_mode=True or False, **kwargs)
+        self.call_hook('after_train(val)_iter')
```

Comment:
This part of code is a little bit vague `train(val)`. Use pseudo-code if necessary.

Reply:
Does it look clearer now?